src/main/java/gift/domain/option/OptionService.java

```diff
@@ -82,4 +85,30 @@ public void updateOption(Long productId, Long optionId, OptionRequestDTO optionR
 
         optionRepository.save(option);
     }
+
+    // 상품 옵션 수량 차감
+    @Transactional
+    public void decreaseOptionQuantity(Long productId, Long optionId, Long quantity) {
+        Product product = productRepository.findById(productId)
+            .orElseThrow(() -> new ProductNotFoundException(productId));
+        Option option = optionRepository.findById(optionId)
+            .orElseThrow(() -> new OptionNotFoundException(optionId));
+
+        if (option.getQuantity() < quantity || quantity < 0) {
+            throw new BusinessException(ErrorCode.BAD_REQUEST, "차감할 수량의 값이 올바르지 않습니다.");
+        }
+
+        if (option.getQuantity() == quantity) {
```

Comment:
이 부분은 객체 비교라 equals로 해주어야 할 것 같습니다!
```suggestion
        if (Objects.equals(option.getQuantity(), quantity)) {
```

Reply:
수정하겠습니다!